rfcs/20230323-add-split-graph-op/README.md

```diff
@@ -0,0 +1,161 @@
+| attribute | `size_splits` | list of sizes for each output alone axis dims.                                         | optional             | s64           |
+| output    | `dst_i`       | output data                                                                            | Required             | f32,f16,bf16* |
+
+@note: 
+- input and outputs must have the same data type.
+-The i-th output has the same shape as src input tensor except for dimension
+along `axis` which is src.dims[`axis`]/`num_splits` or `size_splits`[i].
+- Either `num_splits` or `size_splits` should be provided. When `num_splits` is 
+used,`size_splits` will be ignored.
+- If `num_splits` is specified, the dimension of input tensor data shape along 
+`axis` must be evenly divisible by `num_splits`.
+- If `size_splits` is specified, the sum of the values in `size_splits` must be 
+equal to the src.dims[`axis`].
+
+Pros.
+1. The definition is simple and we only focus on the static shape support.
+
+Cons.
+1. We may need to add a dynamic split in the future when graph API support dynamic shape.
+2. Not suitable for Pytorch integration to map `split_size_or_sections ` of `Split`
+in Pytorch to `num_splits`.
+
+### Option2: 
+
+`Split` operation supports four attributes: `axis`, `num_splits` , `size_split`
+and `size_splits`. `axis`, `num_splits` and `size_split` should be an interger 
+value. `size_splits` is a 1-D Tensor (or list). The input data is split into 
+`num_splits` or len(`size_splits`) elements. The shape of the i-th element has 
+the same size as the value except along dimension `axis` where the size is 
+determined by one of `num_splits`, `size_split`, `size_splits`. Note that Split
+operation is not guaranteed to return a view or a copy of original input tensor.
+
+| Split     | Argument Name | Description                                                                            | Required or Optional | Value Type    |
+|-----------|---------------|----------------------------------------------------------------------------------------|----------------------|---------------|
+| input     | `src`         | input data                                                                             | Required             | f32,f16,bf16* |
+| attribute | `axis`        | Specifies which dimension to split along.                                              | Required             | s64           |
+| attribute | `num_splits`  | number of outputs into which the input tensor data will be split along axis dimension. | optional             | s64           |
+| attribute | `size_split`  | size for single output tensor along axis dimension.                                    | optional             | s64           |
+| attribute | `size_splits` | list of sizes for each output alone axis dimension.                                    | optional             | s64           |
```

Comment:
The naming of `size_splits` and `size_split` are much similar, but they have very different semantics as you note below. It may be confuse users, suggest to give a more descriptive naming.

Reply:
ok, what about `sections_split` just similar to PyTorch.